tests/test_dm_control.py

```diff
@@ -0,0 +1,139 @@
+"""Tests the functionality of the DMEnvWrapper on dm_control envs."""
+import warnings
+from typing import Callable
+
+import dm_control.suite
+import dm_env
+import gymnasium as gym
+import numpy as np
+import pytest
+from dm_control.suite.wrappers import (
+    action_noise,
+    action_scale,
+    mujoco_profiling,
+    pixels,
+)
+from gymnasium.error import Error
+from gymnasium.utils.env_checker import check_env, data_equivalence
+
+from shimmy.dm_control_compatibility import DmControlCompatibility
+from shimmy.registration import DM_CONTROL_ENVS
+
+
+def test_dm_control_envs():
+    """Tests that all DM_CONTROL_ENVS are equal to the known dm-control.suite tasks."""
+    assert dm_control.suite.ALL_TASKS == DM_CONTROL_ENVS
+
+
+CHECK_ENV_IGNORE_WARNINGS = [
+    f"\x1b[33mWARN: {message}\x1b[0m"
+    for message in [
+        "A Box observation space minimum value is -infinity. This is probably too low.",
+        "A Box observation space maximum value is -infinity. This is probably too high.",
+        "For Box action spaces, we recommend using a symmetric and normalized space (range=[-1, 1] or [0, 1]). See https://stable-baselines3.readthedocs.io/en/master/guide/rl_tips.html for more information.",
+        "A Box observation space has an unconventional shape (neither an image, nor a 1D vector). We recommend flattening the observation to have only a 1D vector or use a custom policy to properly process the data. Actual observation shape: ()",
+        "A Box observation space has an unconventional shape (neither an image, nor a 1D vector). We recommend flattening the observation to have only a 1D vector or use a custom policy to properly process the data. Actual observation shape: (8, 2)",
+        "A Box observation space has an unconventional shape (neither an image, nor a 1D vector). We recommend flattening the observation to have only a 1D vector or use a custom policy to properly process the data. Actual observation shape: (2, 4)",
+        "A Box observation space has an unconventional shape (neither an image, nor a 1D vector). We recommend flattening the observation to have only a 1D vector or use a custom policy to properly process the data. Actual observation shape: (4, 4)",
+    ]
+]
+
+
+@pytest.mark.parametrize("domain_name, task_name", DM_CONTROL_ENVS)
+def test_check_env(domain_name, task_name):
+    """Check that environment pass the gymnasium check_env."""
+    env = gym.make(f"dm_control/{domain_name}-{task_name}-v0")
+
+    with warnings.catch_warnings(record=True) as caught_warnings:
```

Comment:
Could we also test that the core wrapper works without wrapping? This is important cause dm_env allows user-defined environments as well.

Reply:
It sort of did that with the dm-control wrappers at the end